code/frameworks/nextjs/src/headers/cookies.ts

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-underscore-dangle */
+import { fn } from '@storybook/test';
+import { headers } from './headers';
+import {
+  parseCookie,
+  stringifyCookie,
+  type RequestCookie,
+} from 'next/dist/compiled/@edge-runtime/cookies';
+
+// Mostly copied from https://github.com/vercel/edge-runtime/blob/c25e2ded39104e2a3be82efc08baf8dc8fb436b3/packages/cookies/src/request-cookies.ts#L7
+class CookieStore {
```

Comment:
shall we implement the interface of Next, to make sure the types are compatible?